### DISCUSSION TITLE
Why do we need to create a copy of the argument to avoid side-effects?

### DISCUSSION THREAD

**Discussion by alexander-lopez-s** [lab-brussels-1/Group4-Practice-code-review#1] [General]
I created my first `arrow function` (I prefer the conventional syntax), and would like to know why do we need to back up our arguments.  I read that it avoids any side effects, but what effects specifically? 

This is my function and it works 100%:

```
const sortingNumbers = (price = []) => {
  const backingUp = [...price];
  backingUp.sort(function (a, b) {
    return (a - b);
  });
  return backingUp;
};

```

@danielhalasz @colevandersWands

**Comment by danielhalasz** [comment]
well, it is a good idea to keep the original array untouched.. that the function is clean and does not modify it..but instead returns a new array with the intended modifications.

**Reply by alexander-lopez-s** [reply]
This makes total sense, in case we need the original array!!
